app/services/summary_service.rb

```diff
@@ -64,13 +64,14 @@ def self.summary(summary_type, prison, page, params)
 
     # For the allocated offenders, we need to provide the allocated POM's
     # name
+    offender_items = bucket.take(wanted_items, from) || []
+
     if summary_type == :allocated
-      offender_items = OffenderService.set_allocated_pom_name(
-        bucket.take(wanted_items, from) || [],
-        prison
-      )
-    else
-      offender_items = bucket.take(wanted_items, from) || []
+      offender_items.each { |offender|
+        alloc = active_allocations_hash[offender.offender_no]
+        offender.allocated_pom_name = alloc.primary_pom_name.titleize
+        offender.allocation_date = alloc.primary_pom_allocated_at
```

Comment:
Do we know that alloc.primary_pom_allocated_at is set for all allocations?  What should we use otherwise?

Reply:
Looks like this is already using this in the current code and so it's fine. 

For the record, this drops the number of SQL queries in the allocated summary from 37 to 11 which is a useful and significant drop.